### PR TITLE
[Linter] Checks for functions with too many parameters.

### PR DESCRIPTION
**Pull request by dzungdinh94**
## Description 

Checks for functions with too many parameters in Move code. Functions that exceed a certain number
of parameters are flagged to encourage better modularity and design practices.

# Run the linter
cargo test move_check_testsuit

# Testing
File testing: too_many_arguments.move

### CODE REVIEW

**Review by tnowacki — CHANGES_REQUESTED** [review]
The concept is good, but a bit strange for Move for two reasons:
1. As a whole the lint isn't really actionable, because we do not have public struct construction. In normal languages, this sort lint exists as warning that you may want to encapsulate your data. But this sort of pattern is not always more ergonomic in Move. 
2. We actually have hard limits for these sort of things on the Sui layer. We should probably integrate those limits into the Sui layer a bit more closely, like we do for the `init` and `entry` checks for example. 

In other words, there is a good compiler check for Sui here, but probably not a good lint.
